src/header-bidding/prebid/appnexus.spec.ts

```diff
@@ -0,0 +1,209 @@
+	window.guardian.config.page.section = 'Magic';
+	window.guardian.config.page.isDev = false;
+};
+
+describe('getAppNexusDirectPlacementId', () => {
+	afterEach(() => {
+		jest.resetAllMocks();
+		resetConfig();
+	});
+
+	test('should return correct placementID for any existing biding sizes in Australia or New Zealand', () => {
+		isInAuOrNz.mockReturnValue(true);
+		getBreakpointKey.mockReturnValue('D');
+		containsMpu.mockReturnValue(true);
+		expect(getAppNexusDirectPlacementId([createAdSize(300, 250)])).toBe(
+			'11016434',
+		);
+	});
+
+	test('should return correct placementID for mobile-sticky and in ROW', () => {
+		isInRow.mockReturnValue(true);
+		getBreakpointKey.mockReturnValue('M');
+		containsMobileSticky.mockReturnValue(true);
+		expect(getAppNexusDirectPlacementId([createAdSize(320, 50)])).toBe(
+			'31512573',
+		);
+	});
+
+	test('should return correct placementID for ad sizes not in the conditional statement and not mobile-sticky in ROW', () => {
+		isInRow.mockReturnValue(true);
+		getBreakpointKey.mockReturnValue('D');
+		containsDmpu.mockReturnValue(true);
+		expect(getAppNexusDirectPlacementId([createAdSize(300, 600)])).toBe(
+			'9251752',
+		);
+	});
+
+	test('should return correct placementID for desktop billboard not leaderboad in UK, ROW and US', () => {
+		isInUk.mockReturnValue(true);
+		isInRow.mockReturnValue(true);
+		isInUsa.mockReturnValue(true);
+		getBreakpointKey.mockReturnValue('D');
+		containsBillboardNotLeaderboard.mockReturnValue(true);
+		expect(getAppNexusDirectPlacementId([createAdSize(970, 250)])).toBe(
+			'30017511',
+		);
+	});
+
+	test('should return correct placementID for desktop mpu or dmpu in UK, ROW and US', () => {
+		isInUk.mockReturnValue(true);
+		isInRow.mockReturnValue(true);
+		isInUsa.mockReturnValue(true);
+		getBreakpointKey.mockReturnValue('D');
+		containsMpuOrDmpu.mockReturnValue(true);
+		expect(getAppNexusDirectPlacementId([createAdSize(300, 600)])).toBe(
+			'9251752',
+		);
+		expect(getAppNexusDirectPlacementId([createAdSize(300, 250)])).toBe(
+			'9251752',
+		);
+	});
+
+	test('should return correct placementID for desktop billboard or leaderboard in UK, ROW and US', () => {
+		isInUk.mockReturnValue(true);
+		isInRow.mockReturnValue(true);
+		isInUsa.mockReturnValue(true);
+		getBreakpointKey.mockReturnValue('D');
+		containsLeaderboardOrBillboard.mockReturnValue(true);
+		expect(getAppNexusDirectPlacementId([createAdSize(970, 250)])).toBe(
+			'9926678',
+		);
+		expect(getAppNexusDirectPlacementId([createAdSize(728, 90)])).toBe(
+			'9926678',
+		);
+	});
+
+	test('should return correct placementID for desktop leaderboard in UK, ROW and US', () => {
+		isInUk.mockReturnValue(true);
+		isInRow.mockReturnValue(true);
+		isInUsa.mockReturnValue(true);
+		getBreakpointKey.mockReturnValue('D');
+		containsLeaderboard.mockReturnValue(true);
+		expect(getAppNexusDirectPlacementId([createAdSize(728, 90)])).toBe(
+			'9251752',
```

Comment:
I would expect this placementID to be 9926678. `containsLeaderboardOrBillboard` would be true if the ad is a leaderboard size, so it should be the same as the test above.

Reply:
I get your point here because of the breakpoint desktop but when I run the tests it gives me the placementID for tablet for `containsLeaderboard` function. Is that because I use `containsLeaderboard`? There is something not quite right about it.

Reply:
Based on our in person chat I added `containsBillboard` to be be false but it gives me the same error and it expects it to be`9251752` which doesn't make sense because this is a placementID for tablet. Not sure what to try next 😅

Reply:
I would do something like this 
```
containsLeaderboard.mockReturnValue(true);
containsBillboard.mockReturnValue(false);
containsLeaderboardOrBillboard.mockReturnValue(
    containsBillboard() || containsLeaderboard(),
);
```
That should give you the correct placement ID 🙂

Reply:
That actually worked. I wouldn't think about using `containsLeaderboardOrBillboard` function for this test but I understand now why it has to be used. Done in 12648f9372f9fb09dbe7a793a47d6c6bdbe748aa